### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by agisboye** [lawtalker/R972_Serial#1]
Hi,

I was delighted to stumble upon this repo. I'm looking to control a R-872 over serial, which I think will be similar to the R-972.

How did you figure out the encoding of the commands? Do you have any documentation on how to communicate with the receiver?

Any help and information is appreciated. I hope to hear from you. :-)

**Comment by lawtalker** [comment]
The short version is that the commands can be sent over serial in the form aa 05 05 XX CK where XX is a command code and CK is checksum. As it works out, CK == 4b for command code 01, and as the command code is incremented, the checksum gets decremented.

More details, including the actual command codes, are in an [Excel file](https://drive.google.com/file/d/1egPDY9d7PazukHOkZJgrQdRNcCKnVRi5/view?usp=sharing) that I got from Sherwood years ago.

If you want to read state (current volume level, etc. - basically, all the things you can see on the face panel of your receiver), you send an event request (command code ff, checksum 04d), and then I think that several bytes of state are sent back to you. I used to have some documentation for that, but I don't know that I have that anymore, and I never got around to implementing anything that used that. It's possible that I got that documentation somewhere on AVS Forum, so you could poke around there.

**Comment by agisboye** [comment]
Thanks! That spreadsheet is super helpful.

I poked around and found a link to a [document describing the protocol](https://web.archive.org/web/20080723134626/http://www.sherwoodusa.com/pdf/SherwoodRS232protocol(7th).pdf). Fortunately, the Internet Archive has actually cached it, since the Sherwood website appears to have been taken down. Is this the document you were thinking of?

It's for the previous generation of receivers and someone noted that the codes don't work with the Rx72 models. Hopefully the rest is unchanged.

**Comment by lawtalker** [comment]
Yes, that looks familiar - either that is the doc I had, or it's at least very similar. (Also handily notes that the serial connection is 8N1.)
